src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.cs

```diff
@@ -266,6 +266,32 @@ public ValueTask HandleAsync(ProcessAuthenticationContext context)
                 }
             }
 
+            else if (context.Registration.ProviderType is ProviderTypes.Huawei)
+            {
+                var error = (string?) context.Request[Parameters.Error];
```

Comment:
Since Huawei only seems to return numeric codes, a `(long?) context.Request[Parameters.Error]` conversion + an `if (error is not null)` check could be used here.